### PR TITLE
Release 0.3.0

### PR DESCRIPTION
**Pull request by glatterf42**
This PR prepares for release 0.3.0 by updating the GitHub Action workflows to test python 3.10 and 3.11. `pyproject.toml` already allows using 3.11. 

It also introduces two workflows to check black and isort formatting on PRs.

### CODE REVIEW

**Comment by danielhuppmann** [comment]
Please update the badge in the readme and the docs-index-page (and maybe other places) to show that Python 3.11 is also supported.

File: .github/workflows/build-docs.yaml [hunk 1]
```diff
@@ -8,6 +8,12 @@ on:
 
 jobs:
   build-docs:
+    strategy:
+      matrix:
+        python-version:
+        - "3.10"  # Earliest version supported by ixmp4
```
Comment (danielhuppmann) [inline .github/workflows/build-docs.yaml:14]: Building the docs only needs to be checked with the version that is actually used on readthedocs.

File: .github/workflows/isort.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
+# copied from https://github.com/marketplace/actions/install-poetry-action
```
[the remark this answers is not in the document]
Comment (glatterf42) [reply .github/workflows/isort.yaml:1]: Yes, running all workflows on both versions is not necessary. 
We could also combine isort, black, flake8 and mypy to a single workflow, as is done for message_ix etc.

[the remark this answers is not in the document]
Comment (danielhuppmann) [reply .github/workflows/isort.yaml:1]: Yes, that's preferable - no need to waste computing resources...

File: .github/workflows/build-docs.yaml (hunk 1 above)
Comment (danielhuppmann) [reply .github/workflows/build-docs.yaml:14]: Yes...

https://github.com/iiasa/ixmp4/blob/27b3000facd3fa96b278ebe471e429d98ee64c16/.readthedocs.yaml#L12

Comment (glatterf42) [reply .github/workflows/build-docs.yaml:14]: And do we want to keep that or try the more future-proof version?

Comment (danielhuppmann) [reply .github/workflows/build-docs.yaml:14]: Fine to keep this until 3.10 isn't supported any more by ixmp4 - my point was just to minimize the computational load.

File: README.md [hunk 3]
```diff
@@ -4,6 +4,7 @@ Copyright (c) 2023 IIASA - Energy, Climate, and Environment Program (ECE)
 
 [![license: MIT](https://img.shields.io/badge/license-MIT-brightgreen.svg)](https://github.com/iiasa/ixmp4/blob/main/LICENSE)
 [![python](https://img.shields.io/badge/python-3.10-blue?logo=python&logoColor=white)](https://github.com/iiasa/ixmp4)
+[![python](https://img.shields.io/badge/python-3.11-blue?logo=python&logoColor=white)](https://github.com/iiasa/ixmp4)
```
Comment (danielhuppmann) [inline README.md:7]:
```suggestion
[![python](https://img.shields.io/badge/python-3.10_|_3.11-blue?logo=python&logoColor=white)](https://github.com/iiasa/ixmp4)
```

File: doc/source/index.rst [hunk 4]
```diff
@@ -14,6 +14,9 @@ Copyright © 2023 IIASA - Energy, Climate, and Environment Program (ECE)
 .. |python| image:: https://img.shields.io/badge/python-3.10-blue?logo=python&logoColor=white
    :target: https://github.com/iiasa/ixmp4
 
+.. |python| image:: https://img.shields.io/badge/python-3.11-blue?logo=python&logoColor=white
+   :target: https://github.com/iiasa/ixmp4
+
```
Comment (danielhuppmann) [inline doc/source/index.rst:19]:
```suggestion
.. |python| image:: https://img.shields.io/badge/python-3.10_|_3.11-blue?logo=python&logoColor=white
   :target: https://github.com/iiasa/ixmp4

```

**Review by danielhuppmann — COMMENTED** [review]
Looks nicer to include the python version in one badge...

File: README.md (hunk 3 above)
Comment (glatterf42) [reply README.md:7]: Thanks...

File: doc/source/index.rst (hunk 4 above)
Comment (glatterf42) [reply doc/source/index.rst:19]: ...and thanks again :)

**Comment by glatterf42** [comment]
The tests are passing and, with the addition of build to the dev dependency group, step 2 of the [release procedure](https://github.com/iiasa/ixmp4/blob/release/0.3.0/DEVELOPING.md#release-procedure) is also successful. Once we merge this PR, we can tag the candidate on main and continue with the release.

**Review by danielhuppmann — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**